daml-foundations/daml-tools/daml-extension/src/extension.ts

```diff
@@ -266,8 +266,12 @@ namespace DamlWorkspaceValidationsNotification {
 }
 
 class VirtualResourceManager {
+    // Mapping from URIs to the web view panel
     private _panels: Map<string, vscode.WebviewPanel> = new Map<string, vscode.WebviewPanel>();
+    // Mapping from URIs to the HTML content of the webview
     private _panelContents: Map<string, string> = new Map<string, string>();
+    // Mapping from URIs to selected view
+    private _panelStates: Map<string, string> = new Map<string, string>();
```

Comment:
Selected view as a `string` is also a bit sad.

Reply:
Given that we treat it as an opaque value, i.e., we just take what the client gave us and send it back I don’t think it makes sense to actually parse this value in particular given the backwards compat constraints on the IDE. In Haskell i would have used a newtype but searching for newtypes in typescript didn’t bring up anything that looked reasonably nice.

Reply:
I think type aliases for uri and html would go a long way to making the comments part of the code, even if they aren't statically checked